staging/src/k8s.io/cloud-provider/types.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloudprovider
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// KubeCloudSharedConfiguration contains elements shared by both kube-controller manager
+// and cloud-controller manager, but not genericconfig.
+type KubeCloudSharedConfiguration struct {
```

Comment:
Our goal should be to being the shared configuration into consistency between the KCM and CCM. I'd rather we spent the time to fix this now before it gets worse.